test/powershell/engine/InitialSessionState.Tests.ps1

```diff
@@ -54,4 +54,90 @@ Describe "InitialSessionState capacity" -Tags CI {
         $ps.AddScript('New-Alias -Name a5000 -Value f1; a5000').Invoke() | Should Be "fn f1"
         $ps.Streams.Error | Should Be $null
     }
-}
\ No newline at end of file
+}
+
+##
+## A reused InitialSessionState created from a TypeTable should not have duplicate types.
+##
+Describe "TypeTable duplicate types in reused runspace InitialSessionState TypeTable" -Tags 'Feature' {
+
+    Context "No duplicate types test" {
+
+        BeforeAll {
+
+            $typeTable = [System.Management.Automation.Runspaces.TypeTable]::new([string[]](Join-Path $PSScriptRoot "../Common/TestTypeFile.ps1xml"))
+            [initialsessionstate] $iss = [initialsessionstate]::Create()
+            $iss.Types.Add($typeTable)
+            [runspace] $rs1 = [runspacefactory]::CreateRunspace($iss)
+
+            # Process TypeTable types from ISS
+            $rs1.Open()
+
+            # Get processed ISS from runspace.
+            $issReused = $rs1.InitialSessionState.Clone()
+            $issReused.ThrowOnRunspaceOpenError = $true
+
+            # Create new runspace with reused ISS.
+            $rs2 = [runspacefactory]::CreateRunspace($issReused)
+        }
+
+        AfterAll {
+
+            if ($rs1 -ne $null) { $rs1.Dispose() }
+            if ($rs2 -ne $null) { $rs2.Dispose() }
+        }
+
+        It "Verifies that a reused InitialSessionState object created from a TypeTable object does not have duplicate types" {
+
+            $errs = $null
+            try
+            {
+                $rs2.Open()
+            }
+            catch
+            {
+                $errs = $_
+            }
+
+            $errs | Should Be $null
+        }
+    }
+
+    Context "Cannot use shared TypeTable in ISS test" {
+
+        BeforeAll {
+
+            # Create default ISS and add shared TypeTable.
+            $typeTable = [System.Management.Automation.Runspaces.TypeTable]::new([string[]](Join-Path $PSScriptRoot "../Common/TestTypeFile.ps1xml"))
+            [initialsessionstate] $iss = [initialsessionstate]::CreateDefault2()
+            $iss.Types.Add($typeTable)
+            $iss.ThrowOnRunspaceOpenError = $true
+            [runspace] $rs = [runspacefactory]::CreateRunspace($iss)
+        }
+
+        AfterAll {
+
+            if ($rs -ne $null) { $rs.Dispose() }
+        }
+
+        It "Verifies that shared TypeTable is not allowed in ISS" {
+
+            # Process TypeTable types from ISS.
+            $errorId = ""
+            try
+            {
+                $rs.Open()
+                throw "No Exception!"
+            }
+            catch
+            {
+                if ($_.Exception.InnerException -ne $null)
+                {
+                    $errorId = $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId
+                }
+            }
+
+            $errorId | Should Be "ErrorsUpdatingTypes"
```

Comment:
I have already been through this with Illya.  The problem with this is that this can result in a null reference exception.  It is better to test whether the desired error Id is found.